File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
    maxCandies := 0
    for _, v := range candies {
        if v > maxCandies {
            maxCandies = v
        }
    }
    
    result := make([]bool, len(candies))
    for i, v := range candies {
        result[i] = v + extraCandies >= maxCandies
    }
    return result
}

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
bool* kidsWithCandies(int* candies, int candiesSize, int extraCandies, int* returnSize){
    int maxc = 0; 
    for (int i = 0; i < candiesSize; i++)
        maxc = (maxc > candies[i]) ? maxc : candies[i];
    
    bool *result = malloc(sizeof(bool) * candiesSize);
    for (int i = 0; i < candiesSize; i++)
        result[i] = (candies[i] + extraCandies) >= maxc;
    
    *returnSize = candiesSize;
    return result;
}